packages/bpk-component-input/src/BpkClearButton.tsx

```diff
@@ -30,14 +31,19 @@ const getClassName = cssModules(STYLES);
 const ClearButtonIcon = withButtonAlignment(ClearIcon);
 
 type Props = {
```

Comment:
I would offer 
```suggestion
interface Props extends ComponentProps<'button'> {
```
since you allow to pass through `...rest` props. Then, you don't need to manually specify `onClick`, `className` (and leave it `undefined`, without defaulting to `null`)

Reply:
No strong opinion for `type` vs `interface` for me, `banana` is happy with `type`, but there are some [hints here](https://react-typescript-cheatsheet.netlify.app/docs/basic/getting-started/basic_type_example/#types-or-interfaces). 

For me, in component libraries, you often need to extend and override some types. With `type`, TypeScript will silently merge the types in whatever way it thinks is best, with `interface` it will complain if handling of the same properties is not clear. [Short example](https://www.typescriptlang.org/play?#code/C4TwDgpgBAglC8UDeAoK6qkgLigcgCMBXYYAewDs8UBfFFLaAIQWTXUdwGdgAnASwoBzANy16jKAGFWLAGSwxKAMaUeUYqUoAFXmTBdcMxKgyZwEXHgAmXAIa271OikHAIvAGZ3l0GATYzTnxNcipxVwp3Lx9mANMOC24+QVEItw9vX2kAiAAPdwpbWAIAGigmeJdVCnVQnT0DJiMAk3ZEnHxHR2d6IA)

Reply:
Good point and thanks for the example, yeah I think that makes sense - will change it to an interface!